File: superHeroInfoAppwithJava/app/src/main/java/com/abdbzkn/superheroinfoappwithjava/DetailsActivity.java
package com.abdbzkn.superheroinfoappwithjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.abdbzkn.superheroinfoappwithjava.databinding.ActivityDetailsBinding;
import com.abdbzkn.superheroinfoappwithjava.databinding.ActivityMainBinding;
import org.w3c.dom.Text;

public class DetailsActivity extends AppCompatActivity {

    private ActivityDetailsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
       binding = ActivityDetailsBinding.inflate(getLayoutInflater());
       View view = binding.getRoot();
       setContentView(view);

       Intent intent = getIntent();
       Superhero selectedSuperhero = (Superhero) intent.getSerializableExtra("superhero");
       binding.superHeroNameText.setText(selectedSuperhero.name);
       binding.superHeroInfoText.setText(selectedSuperhero.info);
       binding.imageView.setImageResource(selectedSuperhero.image);
    }
}

File: superHeroInfoAppwithKotlin/app/src/main/java/com/abdbzkn/superheroinfoappwithkotlin/DetailsActivity.kt
package com.abdbzkn.superheroinfoappwithkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.abdbzkn.superheroinfoappwithkotlin.databinding.ActivityDetailsBinding

class DetailsActivity : AppCompatActivity() {

    private lateinit var binding : ActivityDetailsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDetailsBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        val intent = intent
        //casting yapıyoruz as ile
        //val selectedSuperhero = intent.getSerializableExtra("superhero") as Superhero intent ile almiycam
        val selectedSuperhero = MySingleton.chosenSuperhero

        selectedSuperhero?.let{
            binding.nameText.text = it.name
            binding.infoText.text = it.info
            binding.imageView.setImageResource(it.image)
        }
    }
}